index.js

```diff
@@ -113,65 +105,45 @@ var getImages = (function() {
         imageRegex.lastIndex = 0;
 
         // remove nulls and duplicates
-        images = _.chain(images)
-            .filter()
-            .unique(function(image) {
-                return image.path;
-            })
-            .value();
+        images = _.uniq(_.filter(images), function(image) {
+            return image.path;
+        });
 
-        return Q(images)
+        return Promise.resolve(images)
             // apply user filters
             .then(function(images) {
-                return Q.Promise(function(resolve, reject) {
-                    async.reduce(
-                        options.filter,
-                        images,
-                        function(images, filter, next) {
-                            async.filter(
-                                images,
-                                function(image, ok) {
-                                    Q(filter(image)).then(ok);
-                                },
-                                function(images) {
-                                    next(null, images);
-                                }
-                            );
-                        },
-                        function(err, images) {
-                            if (err) {
-                                return reject(err);
-                            }
-
-                            resolve(images);
-                        }
-                    );
+                return new Promise(function(resolve, reject) {
+                    async.filter(images, function(image, callback) {
+                        async.reduce(options.filter, true, function(status, filter, callback) {
+                            if (!status) return callback(null, false);
```

Comment:
Could you wrap `return callback...` in a curly brackets? It will be more readable, I think.